src/openvic-simulation/GameManager.cpp

```diff
@@ -8,19 +14,103 @@ GameManager::GameManager(
+size_t current_recurse;
+bool GameManager::_get_mod_dependencies(Mod const* mod, std::vector<Mod const*>& dep_list) {
+	bool ret = true;
+
+	for (std::string_view dep_identifier : mod->get_dependencies()) {
+		if (!mod_manager.has_mod_identifier(dep_identifier)) {
+			Logger::error("Mod \"", mod->get_identifier(), "\" has unmet dependency \"", dep_identifier, "\" and cannot be loaded!");
+			return false;
+		}
+
+		Mod const* dep = mod_manager.get_mod_by_identifier(dep_identifier);
+		/* The poor man's cycle checking (cycles should be very rare and hard to accomplish with vic2 modding, this is a failsafe)*/
+		if (current_recurse == MAX_RECURSE) {
+			Logger::error("Mod \"", mod->get_identifier(), "\" has cyclical or broken dependency chain and cannot be loaded!");
+			return false;
+		} else {
+			current_recurse++;
+			ret &= _get_mod_dependencies(dep, dep_list); /* recursively search for mod dependencies */
+		}
+		if (std::find(dep_list.begin(), dep_list.end(), dep) == dep_list.end()) {
+			dep_list.emplace_back(dep);
+		}
+	}
+
+	return ret;
+}
+
+bool GameManager::load_mods(Dataloader::path_vector_t& roots, Dataloader::path_vector_t& replace_paths, std::vector<std::string> const& requested_mods) {
+	bool ret = true;
+
+	std::vector<Mod const*> load_list;
+
+	/* Check loaded mod descriptors for requested mods, using either full name or user directory name
+	 * (Historical Project Mod 0.4.6 or HPM both valid, for example), and load them plus their dependencies.
+	 */
+	for (std::string_view requested_mod : requested_mods) {
+		auto it = std::find_if(
+			mod_manager.get_mods().begin(),
+			mod_manager.get_mods().end(),
+			[&requested_mod](Mod const& mod) -> bool {
+				return mod.get_identifier() == requested_mod || mod.get_user_dir() == requested_mod;
+			}
+		);
+
+		if (it == mod_manager.get_mods().end()) {
+			Logger::error("Requested mod \"", requested_mod, "\" does not exist!");
+			ret = false;
+			continue;
+		}
+
+		std::vector<Mod const*> dependencies;
+		current_recurse = 0;
+		if(!_get_mod_dependencies(&*it, dependencies)) {
+			ret = false;
+			continue;
+		}
+
+		/* Add mod plus dependencies to load_list in proper order. */
+		load_list.reserve(1 + dependencies.size());
+		for (Mod const* dep : dependencies) {
+			if (ret && std::find(load_list.begin(), load_list.end(), dep) == load_list.end()) {
+				load_list.emplace_back(dep);
+			}
+		}
+		if (ret && std::find(load_list.begin(), load_list.end(), &*it) == load_list.end()) {
+			load_list.emplace_back(&*it);
+		}
+	}
+
+	/* Actually registers all roots and replace paths to be loaded by the game. */
+	for (Mod const* mod : load_list) {
+		roots.emplace_back(roots[0] / mod->get_dataloader_root_path());
+		for (std::string_view path : mod->get_replace_paths()) {
+			if (std::find(replace_paths.begin(), replace_paths.end(), path) == replace_paths.end()) {
+				replace_paths.emplace_back(path);
+			} 
+		}
+	}
+	mod_manager.set_loaded_mods(load_list);
```

Comment:
load_list is essentially dead after this, you're performing a copy here, it should only ever move.